### PR TITLE
Oscar - Eliminates Restaurant, UCSBDates frontend components

### LINKED ISSUE #18: Eliminate Restaurant/UCSBDates Frontend

### PR DESCRIPTION
**Pull request by obenedek20**
Closes #18 

Merge after #38 

chromatic: https://ucsb-cs156-f24.github.io/proj-rec-f24-08/prs/42/chromatic

Users should see all Navbar and other UI references to restaurants and ucsbdates now removed/inaccessable

Since not already merged, Christy's branch to create CRUD endpoints and mine to remove the backend items are merged to this one

### CODE REVIEW

**Review by divyanipunj — COMMENTED** [review]
Everything looks good to me, but I think RecomendationRequestController and RecomendationRequestControllerTest should reflect the changes that Christy recently made. I'm not entirely sure but if this PR gets merged after Christy's PR gets merged, won't it replace the code that she wrote? I could be wrong - let me know what you think. 

I also left this comment on your other PR (#41).

**Comment by obenedek20** [comment]
> Everything looks good to me, but I think RecomendationRequestController and RecomendationRequestControllerTest should reflect the changes that Christy recently made. I'm not entirely sure but if this PR gets merged after Christy's PR gets merged, won't it replace the code that she wrote? I could be wrong - let me know what you think.
> 
> I also left this comment on your other PR (#41).

Just pulled from Christy's so should be completely up to date!

**Review by divyanipunj — DISMISSED** [review]

File: frontend/src/tests/components/Nav/AppNavbar.test.js [hunk 1]
```diff
@@ -205,26 +199,7 @@ describe("AppNavbar tests", () => {
...
-        </MemoryRouter>
-      </QueryClientProvider>,
-    );
-
-    await screen.findByText("Log In");
-    expect(screen.getByText("Log In")).toHaveAttribute(
-      "href",
-      "/oauth2/authorization/google",
-    );
+    // expect(screen.getByText("Welcome,")).toBeInTheDocument();
```
Comment (pconrad) [inline frontend/src/tests/components/Nav/AppNavbar.test.js:202]: Commented out code should not be in code that is being merged into main.

```suggestion
```

**Review by pconrad — CHANGES_REQUESTED** [review]
Ok… so #41 is all entangled with #39… which needs a major refactor… in ways that are unnecessary.   I’ve put comments on the PR showing a better way to get to your goal so that this PR can be independent of the others. (edited) 




Similarly, this PR is entangled with #41, which is also unnecessary.  If you remove the backend without first removing the frontend for some part of the app, lots of things break (menus and features no longer work).  But it’s fine to remove frontend features without removing the backend… So why, why, why does this PR depend on backend changes?

If it were a clean PR that had started from a fresh copy of main (which should always be your default), it would be mergeable right now (or at least once the commented out code is addressed.)  Instead, in the current form, it’s going to have to wait on a lot of other complicated events.

I suggest redoing it as a clean PR that only removes the frontend code and does nothing else.  That would be mergeable immediately (if it didn't have commented out code.)

**Pull request closed without merging**